Compiler/SimCode/SimCode.mo

```diff
@@ -292,6 +293,37 @@ uniontype DaeModeData
   end DAEMODEDATA;
 end DaeModeData;
 
+uniontype OMSIData
+  "contains data for code generation for OMSI"
+  record OMSI_DATA
+    OMSIFunction initialization "contains equations and variables for initialization problem";
+    OMSIFunction simulation "contains equations and variables for simulation problem";
+  end OMSI_DATA;
+end OMSIData;
+
+uniontype OMSIFunction
+  "contains equations and variables for initialization or simulation problem"
+  record OMSI_FUNCTION
```

Comment:
The whole initialization problem and simulation problem are each one `OMSIFunction`.
Algebraic loops have residual functions, also saved as `OMSIFunction`.

The idea is, that in each case we want to evaluate some function (right hand side of an ODE or evaluate a residual) and `OMSIFunction` got the needed equations (with possible loops) and variables for that.

Reply:
I see! That means, `OMSI_FUNCTION` is used as an abstract representation of an algebraic system of equations with given knowns and unknowns – which is essentially an algorithm, right?

I've one more question, since I haven't worked on the SimCode phase for quite some time now: Are all the equations (`SimEqSystem`) already causalized?

Reply:
Yes, the equations in OMSIFunction.equations are causalized.
It happens in [SimCodeUtil.mo](https://github.com/OpenModelica/OMCompiler/pull/2937/files#diff-b7691fe179103a8a890bf16a8f1c2975R3846) in function `createAllEquationOMSI`.

And in a sense that's an algorithm.

Reply:
Thanks. It should probably also be clarified in the inline documentation.